src/components/invest/index.js

```diff
@@ -386,17 +390,27 @@ export class Invest extends React.Component {
               <div className="invest-form-label">TOKENS</div>
               {invalidTokenDescription}
             </div>
-            <a className="button button_fill" onClick={this.investToTokens}>Invest now</a>
+            <div className="invest-through-container">
+              <select className="invest-through" onChange={(e) => this.setState({ investThrough: e.target.value })}>
+                <option value={INVESTMENT_OPTIONS.metamask}>Metamask</option>
+                <option value={INVESTMENT_OPTIONS.qr}>QR</option>
+              </select>
+              <a className="button button_fill" onClick={this.investToTokens}>Invest</a>
+            </div>
             <p className="description">
             Think twice before investment in ICOs. Tokens will be deposited on a wallet you used to buy tokens.
             </p>
           </form>
+          {
+            this.state.investThrough === INVESTMENT_OPTIONS.qr ?
```

Comment:
please extract 
```
this.state.investThrough === INVESTMENT_OPTIONS.qr ? 
<InvestThroughQR crowdsaleAddress={this.state.crowdsaleAddress} /> :  null
```
to a variable and render that variable

Reply:
As far as I know, you can't do that in a template, can you? I mean, the code inside the template must be an expression, so you can use assignments.

The alternative is to move the expression to a function, and do:

```
renderQRPaymentProcess() {

{ this.renderQRPaymentProcess }
```

Reply:
One can use either return the template in a function or assign it to a variable. Either way will work.